libs/api_common/windows_platform_common.cpp

```diff
@@ -136,10 +428,734 @@ get_attach_type_windows(const std::string& section)
 _Ret_maybenull_z_ const char*
 get_attach_type_name(_In_ const ebpf_attach_type_t* attach_type)
 {
-    // TODO: (Issue #223) Read the registry to fetch attach types.
-    auto it = windows_section_names.find(*attach_type);
-    if (it != windows_section_names.end())
-        return it->second;
+    for (const ebpf_section_definition_t& t : _windows_section_definitions) {
+        if (IsEqualGUID(*t.attach_type, *attach_type)) {
+            return t.section_prefix;
+        }
+    }
+
+    return nullptr;
+}
+
+static ebpf_result_t
+_load_section_data_information(HKEY section_data_key, _In_ const wchar_t* section_name) noexcept
+{
+    int32_t status;
+    ebpf_result_t result = EBPF_SUCCESS;
+    HKEY section_info_key = nullptr;
+    ebpf_program_type_t* program_type = nullptr;
+    ebpf_attach_type_t* attach_type = nullptr;
+    ebpf_section_definition_t section_definition;
+    bpf_prog_type_t bpf_program_type;
+    bpf_attach_type_t bpf_attach_type;
+    char* section_prefix = nullptr;
+
+    try {
+        status = open_registry_key(section_data_key, section_name, KEY_READ, &section_info_key);
```

Comment:
Can this cpp file use ebpf_store_helper.h rather than doing registry operations itself?

Reply:
`ebpf_store_helper.h` only contains store "update" APIs, as those are the only ones needed by kernel mode (I will also add delete_entry APIs in the next PR).

This file is only doing the "read" operations, hence directly calling registry APIs. But agree we can abstract that. 
2 options:
1. Include the "read" APIs also in `ebpf_store_helper.h`. This has 2 problems: first, that all functions in this header will have to be static, even though these extra APIs will not be used by kernel code. second, read APIs will require memory allocation, and will complicate the kernel part of the code.
2. Abstract the "read" APIs in a different _um_only_ file.

Reply:
I didn't follow what "all functions will have to be static" means.  Since the header is just a header, it shouldn't result in any binary impact for things not actually called.   Maybe you're talking about the .c implementation?